File: 0128 longest-consecutive-sequence/solution.rs
use std::collections::HashSet;
impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let mut s = HashSet::new();
        for i in 0..nums.len() {
            s.insert(nums[i]);
        }
        let mut ans = 0;
        for x in &s {
            if !s.contains(&(x - 1)) {
                let mut st = 1;
                let mut curr = *x;
                while s.contains(&(curr + 1)) {
                    st+=1;
                    curr+=1;
                }
                ans = ans.max(st);
            }
        }
        ans
    }
}

File: 0128 longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        ans = 0
        for val in num_set:
            if val - 1 not in num_set:
                curr=val
                len=1
                while curr+1 in num_set:
                    curr+=1
                    len+=1
                ans = max(ans,len)
        return ans
